src/win/fs.c

```diff
@@ -144,26 +144,96 @@ void uv__fs_init(void) {
 }
 
 
+static int32_t fs__decode_wtf8_char(const char** input) {
+  uint32_t code_point;
+  uint8_t b1;
+  uint8_t b2;
+  uint8_t b3;
+  uint8_t b4;
+
+  b1 = **input;
+  if (b1 <= 0x7f)
+    return b1; /* ASCII code point */
+  if (b1 < 0xc2)
+    return -1; /* invalid: continuation byte */
+  code_point = b1;
+
+  b2 = *++*input;
+  if ((b2 & 0xc0) != 0x80)
+    return -1; /* invalid: not a continuation byte */
+  code_point = (code_point << 6) | (b2 & 0x3f);
+  if (b1 <= 0xdf)
+    return 0x7ff & code_point; /* two-byte character */
+
+  b3 = *++*input;
+  if ((b3 & 0xc0) != 0x80)
+    return -1; /* invalid: not a continuation byte */
+  code_point = (code_point << 6) | (b3 & 0x3f);
+  if (b1 <= 0xef)
+    return 0xffff & code_point; /* three-byte character */
+
+  b4 = *++*input;
+  if ((b4 & 0xc0) != 0x80)
+    return -1; /* invalid: not a continuation byte */
+  code_point = (code_point << 6) | (b4 & 0x3f);
+  if (b1 <= 0xf4)
+    if (code_point <= 0x10ffff)
+      return code_point; /* four-byte character */
+
+  /* code point too large */
+  return -1;
+}
+
+
+static ssize_t fs__get_length_wtf8(const char* source_ptr) {
+  size_t w_target_len = 0;
+  int32_t code_point;
+
+  do {
+    code_point = fs__decode_wtf8_char(&source_ptr);
+    if (code_point < 0)
+      return -1;
+    if (code_point > 0xffff)
```

Comment:
Admittedly kind of nitpicky but libuv tends to write hex numbers in capitals, i.e., `0xFFFF`.